### PR TITLE
Feature/kal tab panel

### PR DESCRIPTION
**Pull request by MichaelGrosio**

### CODE REVIEW

File: projects/kalidea/kaligraphi/src/lib/molecules/kal-tabs/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  <kal-tab label="Header 3">
+    Body 3
+  <kal-tab>
+
+</kal-tab-group>
+````
+
+####SPECIFICATIONS
+
+* Doit afficher les onglets en fonction du ng-content
```
Comment (Bellico) [inline projects/kalidea/kaligraphi/src/lib/molecules/kal-tabs/README.md]: French?

File: projects/kalidea/kaligraphi/src/lib/molecules/kal-tabs/kal-tab-group/kal-tab-group.component.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  styleUrls: ['./kal-tab-group.component.sass'],
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class KalTabGroupComponent implements OnInit, AfterContentInit {
+
+  /**
+   * Content to display in the tab body component
+   */
+  contentTemplatePortal: TemplatePortal;
```
Comment (Bellico) [inline projects/kalidea/kaligraphi/src/lib/molecules/kal-tabs/kal-tab-group/kal-tab-group.component.ts]: is it useful ?

File: projects/kalidea/kaligraphi/src/lib/molecules/kal-tabs/kal-tab-group/kal-tab-group.component.html [hunk 3]
```diff
@@ -0,0 +1,24 @@
+<div>
+
+  <h1>HEADER</h1>
```
Comment (Bellico) [inline projects/kalidea/kaligraphi/src/lib/molecules/kal-tabs/kal-tab-group/kal-tab-group.component.html]: Is it the final render?

File: projects/kalidea/kaligraphi/src/lib/molecules/kal-tabs/kal-tab-group/kal-tab-group.component.html [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  </kal-tab-header>
+
+</div>
+
+<div>
+
+  <h1>BODY</h1>
+
+  <kal-tab-body *ngFor="let tab of tabs; index as i"
+                class="test"
```
Comment (Bellico) [inline projects/kalidea/kaligraphi/src/lib/molecules/kal-tabs/kal-tab-group/kal-tab-group.component.html]: Test here ?

File: projects/kalidea/kaligraphi/src/lib/molecules/kal-tabs/kal-tab-body/kal-tab-body.component.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  /**
+   * The reference to the cdk portal outlet
+   */
+  @ViewChild(CdkPortalOutlet) portalOutlet: CdkPortalOutlet;
+
+  constructor(private cdr: ChangeDetectorRef,
+              @Optional() @Inject(forwardRef(() => KalTabGroupComponent)) public tabGroup: KalTabGroupComponent) {
+  }
+
+  ngOnInit() {
```
Comment (jsp-up) [inline projects/kalidea/kaligraphi/src/lib/molecules/kal-tabs/kal-tab-body/kal-tab-body.component.ts]: remove

File: projects/kalidea/kaligraphi/src/lib/molecules/kal-tabs/kal-tab-group/kal-tab-group.component.ts [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+   * Attach a template portal
+   */
+  attachTemplatePortal(portalOutlet: CdkPortalOutlet, content: TemplatePortal, cdr: ChangeDetectorRef) {
+    if (content) {
+      portalOutlet.attachTemplatePortal(content);
+      cdr.detectChanges();
+    }
+  }
+
+  ngOnInit() {
```
Comment (jsp-up) [inline projects/kalidea/kaligraphi/src/lib/molecules/kal-tabs/kal-tab-group/kal-tab-group.component.ts]: remove

File: projects/kalidea/kaligraphi/src/lib/molecules/kal-tabs/kal-tab-header/kal-tab-header.component.ts [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+  styleUrls: ['./kal-tab-header.component.sass'],
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class KalTabHeaderComponent implements OnInit, AfterViewInit {
+
+  /**
+   * The label of the header
+   */
+  @Input() label = '';
```
Comment (jsp-up) [inline projects/kalidea/kaligraphi/src/lib/molecules/kal-tabs/kal-tab-header/kal-tab-header.component.ts:22]: getter and setter ?

File: projects/kalidea/kaligraphi/src/lib/molecules/kal-tabs/kal-tab-header/kal-tab-header.component.ts [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+  get selected() {
+    return this.isSelected;
+  }
+
+  set selected(value: boolean) {
+    this.isSelected = coerceBooleanProperty(value);
+    this.cdr.markForCheck();
+  }
+
+  ngOnInit() {
```
Comment (jsp-up) [inline projects/kalidea/kaligraphi/src/lib/molecules/kal-tabs/kal-tab-header/kal-tab-header.component.ts]: remove

File: projects/kalidea/kaligraphi/src/lib/molecules/kal-tabs/kal-tab/kal-tab.component.ts [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+  styleUrls: ['./kal-tab.component.sass'],
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class KalTabComponent implements OnInit {
+
+  /**
+   * Label of the header
+   */
+  @Input() label = '';
```
Comment (jsp-up) [inline projects/kalidea/kaligraphi/src/lib/molecules/kal-tabs/kal-tab/kal-tab.component.ts:18]: getter and setter ?

File: projects/kalidea/kaligraphi/src/lib/molecules/kal-tabs/kal-tab/kal-tab.component.ts [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+   * Is a tab selected
+   */
+  private selectedTab = false;
+
+  /**
+   * Is a tab disabled
+   */
+  private isDisabled = false;
+
+  constructor() {
```
Comment (jsp-up) [inline projects/kalidea/kaligraphi/src/lib/molecules/kal-tabs/kal-tab/kal-tab.component.ts:40]: remove

File: projects/kalidea/kaligraphi/src/lib/molecules/kal-tabs/kal-tab/kal-tab.component.ts [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+  }
+
+  /**
+   * Return the content of the tab
+   */
+  get content(): TemplatePortal<any> {
+    return this.tabContent;
+  }
+
+  ngOnInit() {
```
Comment (jsp-up) [inline projects/kalidea/kaligraphi/src/lib/molecules/kal-tabs/kal-tab/kal-tab.component.ts]: remove

**Review by jsp-up — CHANGES_REQUESTED** [review]

**Review by jsp-up — APPROVED** [review]

**Pull request merged**